core/aggsigdb/memory_internal_test.go

```diff
@@ -15,14 +14,20 @@ import (
 )
 
 func TestDutyExpiration(t *testing.T) {
-	t.Skip("Disabled due to race condition, see https://github.com/ObolNetwork/charon/issues/2546")
+	t.Parallel()
 
+	var wg sync.WaitGroup
 	ctx, cancel := context.WithCancel(context.Background())
 
 	deadliner := newTestDeadliner()
 	db := NewMemDB(deadliner)
 
-	go db.Run(ctx)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		db.Run(ctx)
+	}()
```

Comment:
```suggestion
	go func() {
		db.Run(ctx)
		wg.Done()
	}()
```
would this work?

Reply:
Using `defer wg.Done()` is a cleaner solution as it ensures that `wg.Done()` will always be called right before the goroutine exits, irrespective of how the function returns or if unexpected panics occur. Placing `wg.Done()` directly at the end depends on the function's flow to reach that point, there's a risk it might not get executed. Using `defer` offers more robustness and clarity in the code's intention.

However, if you believe the direct approach is more appropriate for this example, I respect that viewpoint and it should work in this specific case.

Reply:
I understand & agree with your reasoning, let's do `defer wg.Done()`